src/Sulu/Bundle/LocationBundle/Resources/js/containers/Location/LocationOverlay.js

```diff
@@ -0,0 +1,261 @@
+// @flow
+import React from 'react';
+import {action, observable, reaction, toJS} from 'mobx';
+import {observer} from 'mobx-react';
+import {Form, Input, Number} from 'sulu-admin-bundle/components';
+import Overlay from 'sulu-admin-bundle/components/Overlay';
+import {translate} from 'sulu-admin-bundle/utils/Translator';
+import {Map, Marker, TileLayer} from 'react-leaflet';
+import {SingleAutoComplete} from 'sulu-admin-bundle/containers';
+import type {Location as LocationValue} from '../../types';
+import locationOverlayStyles from './locationOverlay.scss';
+
+type Props = {
+    initialValue: ?LocationValue,
+    onClose: () => void,
+    onConfirm: (?LocationValue) => void,
+    open: boolean,
+};
+
+@observer
+class LocationOverlay extends React.Component<Props> {
+    @observable mapLat: number;
+    @observable mapLong: number;
+    @observable mapZoom: number;
+
+    @observable markerLat: ?number;
+    @observable markerLong: ?number;
+
+    @observable title: ?string;
+    @observable street: ?string;
+    @observable number: ?string;
+    @observable code: ?string;
+    @observable town: ?string;
+    @observable country: ?string;
+
+    updateInitialDataDisposer: () => *;
+
+    constructor(props: Props) {
+        super(props);
+
+        this.updateInitialDataDisposer = reaction(() => this.props.open, (newOpenValue) => {
+            if (newOpenValue === true) {
+                this.mapLat = this.props.initialValue ? this.props.initialValue.lat : 0;
+                this.mapLong = this.props.initialValue ? this.props.initialValue.long : 0;
+                this.mapZoom = this.props.initialValue ? this.props.initialValue.zoom : 1;
+
+                this.markerLat = this.props.initialValue ? this.props.initialValue.lat : null;
+                this.markerLong = this.props.initialValue ? this.props.initialValue.long : null;
+
+                this.title = this.props.initialValue ? this.props.initialValue.title : null;
+                this.street = this.props.initialValue ? this.props.initialValue.street : null;
+                this.number = this.props.initialValue ? this.props.initialValue.number : null;
+                this.code = this.props.initialValue ? this.props.initialValue.code : null;
+                this.town = this.props.initialValue ? this.props.initialValue.town : null;
```

Comment:
Seems like you are already copying over the values anyway, so it's not really only the `initialValue`.

Reply:
would you call this just `value` then?